### ISSUE TITLE
Compatibility with Svelte 5

### ISSUE THREAD

**Issue by metonym** [carbon-design-system/carbon-components-svelte#2012]
Svelte 5 achieved a stable release on October 19th, 2024.

In non-runes mode, Svelte 3/4 libraries are designed to be interoperable with Svelte 5. However, there will be required changes for `carbon-components-svelte`.

This is a parent issue to track issues with using this library with Svelte 5. The short term goal is to make this library compatible with Svelte 5 without introducing breaking changes that impact its current usability with Svelte 3 and 4.

**Comment by metonym** [comment]
These compatibility issues are addressed in [v0.86.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.86.0).

This means that v0.86 is compatible with Svelte 3, 4, and 5 (non-runes mode).

If using Svelte in non-runes mode, the only change you need to make is to rename the `children` prop to `nodes in the `TreeView` and `RecursiveList` components.

```diff
<TreeView
- {children}
+ {nodes}
/>

<RecursiveList
- {children}
+ {nodes}
/>
```

**Comment by theetrain** [comment]
For posterity, this relates to #1877
